lightning/src/routing/network_graph.rs

```diff
@@ -1082,8 +1084,10 @@ impl NetworkGraph {
 				macro_rules! maybe_update_channel_info {
 					( $target: expr, $src_node: expr) => {
 						if let Some(existing_chan_info) = $target.as_ref() {
-							if existing_chan_info.last_update >= msg.timestamp {
+							if existing_chan_info.last_update > msg.timestamp {
 								return Err(LightningError{err: "Update older than last processed update".to_owned(), action: ErrorAction::IgnoreAndLog(Level::Gossip)});
+							} else if existing_chan_info.last_update == msg.timestamp {
+								return Err(LightningError{err: "Update had same timestamp as last processed update".to_owned(), action: ErrorAction::IgnoreDuplicateGossip});
```

Comment:
I don't know if such a comment would be helpful to anybody but me, but I had to look up to verify that the timestamp is strictly monotonic.